items/metadata/metadata.js

```diff
@@ -74,9 +74,45 @@ const removeMetadata = function (itemOrName, namespace) {
   };
 };
 
+// TODO: Move implementation to openHAB Core
```

Comment:
@digitaldan Do you think we should move the underlying implementation of this method to openHAB core?

FY: I decided to use a JS Map instead of an object because IMO Maps are easier to document both in JSDoc and TypeScript than "dynamic" objects. I guess only advanced users will use this method, but it is well documented so Map shouldn't be a problem here.

Reply:
>Do you think we should move the underlying implementation of this method to openHAB core? 

Hmm,  if we think this can be reused by other scripting engines, then maybe.  I don't do much with metadata in my own rules, so i don't have a ton of familiarity with querying/manipulating item metadata

Reply:
> i don't have a ton of familiarity with querying/manipulating item metadata

Same on my side. This PR is more on user request than on personal interest.

Core already has a method to get the namespaces of all metadata on an Item, but this method doesn't deliver the metadata itself.

But I could imagine other script enginges want to use something like this as well.
I would propose to leave the implementation here to get this finished soon, and then I will see if/when I can open a core PR.